skops/tests/test_persist.py

```diff
@@ -0,0 +1,338 @@
+import tempfile
+import warnings
+
+import numpy as np
+import pytest
+from scipy import special
+from sklearn.base import BaseEstimator
+from sklearn.linear_model import LogisticRegression
+from sklearn.pipeline import FeatureUnion, Pipeline
+from sklearn.preprocessing import (
+    FunctionTransformer,
+    MinMaxScaler,
+    PolynomialFeatures,
+    StandardScaler,
+)
+from sklearn.utils import all_estimators
+from sklearn.utils._testing import (
+    SkipTest,
+    assert_allclose_dense_sparse,
+    set_random_state,
+)
+from sklearn.utils.estimator_checks import (
+    _construct_instance,
+    _enforce_estimator_tags_y,
+    _get_check_estimator_ids,
+)
+from sklearn.utils.validation import has_fit_parameter
+
+from skops import load, save
+
+# list of estimators for which we need to write tests since we can't
+# automatically create an instance of them.
+EXPLICIT_TESTS = [
+    "ColumnTransformer",
+    "FeatureUnion",
+    "GridSearchCV",
+    "HalvingGridSearchCV",
+    "HalvingRandomSearchCV",
+    "Pipeline",
+    "RandomizedSearchCV",
+    "SparseCoder",
+]
+
+
+ESTIMATORS_EXPECTED_TO_FAIL_NON_FITTED = {
```

Comment:
The idea is to have one list, where we have the estimator, and when we remove the estimator all tests for that estimator should pass. So I'll revert to the previous list.

Reply:
Hmm, I don't quite understand. What you describe should work with the current approach, no?

One problem of the previous approach is that it does not differentiate between different instantiations of the same estimator. E.g. `FunctionTransformer` with numpy functions works but `FunctionTransformer` with scipy functions doesn't. If only the class is checked, we can't differentiate.

Or is the problem that we have two lists, one for non-fitted and one for fitted?